src/screens/SettingsScreen.tsx

```diff
@@ -42,20 +55,156 @@ export const SettingsScreen: React.FC<SettingsScreenProps> = () => {
   const { prefs, save } = usePrefs()
   const canUseLocalAuth = useCanUseLocalAuth()
   const { user, handleLogout } = useAuth()
+  const { secrets, reset, add } = useSecrets()
+  const [status, requestPermission] = MediaLibrary.usePermissions()
+
+  const backupFileNamePattern = 'optic-backup-{TS}'
+
+  const androidExport = async (fileName, fileContent) => {
+    const permissions =
+      await StorageAccessFramework.requestDirectoryPermissionsAsync()
+    if (!permissions.granted) {
+      Toast.show('Permissions not granted')
+      return
+    }
+
+    try {
+      const uri = await StorageAccessFramework.createFileAsync(
+        permissions.directoryUri,
+        fileName,
+        'application/text'
+      )
+
+      await writeAsStringAsync(uri, fileContent, {
+        encoding: EncodingType.UTF8,
+      })
+      Toast.show('Tokens exported successfully')
+    } catch (err) {
+      Toast.show('An error occurred while exporting tokens')
+      console.log(err)
+    }
+  }
+
+  const iosExport = async (fileName, fileContent) => {
+    const available = Sharing.isAvailableAsync()
+    if (!available) {
+      Toast.show('Sharing not available')
+      return
+    }
+
+    requestPermission()
+
+    if (!status.granted) {
+      Toast.show('Permissions not granted')
+      return
+    }
+
+    const localBackupFileRoute = `${documentDirectory}${fileName}`
+
+    try {
+      await writeAsStringAsync(localBackupFileRoute, fileContent, {
+        encoding: EncodingType.UTF8,
+      })
+
+      await Sharing.shareAsync(localBackupFileRoute, {
+        dialogTitle: fileName,
+        UTI: 'public.item',
+        mimeType: 'text/plain',
+      })
+      Toast.show('Tokens exported successfully')
+    } catch (err) {
+      Toast.show('An error occurred while exporting tokens')
+      console.log(err)
+    }
+  }
+
+  const handleExport = async () => {
+    if (secrets.length === 0) {
+      Toast.show('There are no secrets to export')
+      return
+    }
+
+    const fileName = backupFileNamePattern.replace(
```

Comment:
do we really need to have a pattern with a placeholder? it seems this is the only place where it's used, so we could simply interpolate the string here instead of coming up with a custom replace pattern

Reply:
I used the interpolation as suggested